src/fstests/fstests.jinja2

```diff
@@ -135,30 +141,34 @@ class Job(BaseJob):
         return node
 
     def _run(self, src_path):
-        fail_results = {
+        results = {
             'node': {
                 'name': 'fstests',
-                'result': 'fail',
                 'state': 'done',
             },
             'child_nodes':[],
         }
-        if not self._kernel_config(src_path):
-            return fail_results
-        if not self._kernel_build(src_path):
-            return fail_results
+        ok, result_node = self._kernel_config(src_path)
+        results['child_nodes'].append(result_node)
+        if not ok:
+            return results
+        ok, result_node = self._kernel_build(src_path)
+        results['child_nodes'].append(result_node)
+        if not ok:
+            return results
         if not self._run_smoke_tests():
-            return fail_results
+            return results
         if not self._get_xml_from_vm(src_path):
-            return fail_results
+            return results
```

Comment:
This can be written as a single if statement with `or` operators.

Reply:
@mgalka does python implement the short-circuit? If yes, it will surely stop in the first fail, as expected.

Reply:
You can also consider to use `any` operator.

Reply:
updated to use `or`